query-engine/core/src/query_graph_builder/read/utils.rs

```diff
@@ -259,6 +260,13 @@ pub(crate) fn get_relation_load_strategy(
     nested_queries: &[ReadQuery],
     query_schema: &QuerySchema,
 ) -> QueryGraphBuilderResult<RelationLoadStrategy> {
+    static RELATION_LOAD_STRATEGY: Lazy<Result<RelationLoadStrategy, DomainError>> =
+        Lazy::new(|| std::env::var("RELATION_LOAD_STRATEGY").unwrap().as_str().try_into());
+
+    if RELATION_LOAD_STRATEGY.is_ok() {
+        return Ok(*RELATION_LOAD_STRATEGY.as_ref().unwrap());
```

Comment:
This unfortunately won't help us with creating a test matrix as quite a bunch of tests will be breaking with `join` if it can't actually be used. This will also render the `relationLoadStrategy` argument non-functional.

What this env var actually needs to do is to define the _default_ strategy and not unconditionally override it.

So, instead, you need to change the match statement for `requested_strategy` on `JoinStrategySupport::Yes` below to split the line 281 into this:

```diff
- Some(RelationLoadStrategy::Join) | None => Ok(RelationLoadStrategy::Join)
+ Some(RelationLoadStrategy::Join) => Ok(RelationLoadStrategy::Join),
+ None => Ok(*DEFAULT_RELATION_LOAD_STRATEGY
+     .as_ref()
+     .unwrap_or(&RelationLoadStrategy::Join)),
```

So, in other words, we use the default relation load strategy (whether hardcoded or provided via env var) only when switching before strategies is possible, and no strategy was requested explicitly by the user.